--- a/src/meson.build
+++ b/src/meson.build
@@ -188,6 +188,8 @@ if not spice_gtk_has_version_script
   spice_client_glib_syms = files('spice-glib-sym-file')
   spice_client_glib_syms_path = meson.current_source_dir() / 'spice-glib-sym-file'
   spice_gtk_version_script = ['-export-symbols', spice_client_glib_syms_path]
+  if host_machine.system() == 'darwin'
+    spice_gtk_version_script = []
+  endif
 endif
 
 # soversion
